integrations/terraform/src/index.tsx

```diff
@@ -0,0 +1,109 @@
+import * as jose from 'jose';
+
+import { createComponent, createIntegration, FetchEventCallback } from '@gitbook/runtime';
+
+import { TerraformInstallationConfiguration, TerraformRuntimeContext } from './configuration';
+import { generateJwtSecret, parseJwtSecret } from './token';
+
+const tokenRegExp = /^Bearer\s+(\S+)$/i;
+
+const tokenText = createComponent<{}, {}, void, TerraformRuntimeContext>({
+    componentId: 'jwt',
+    async render(element, context) {
+        element.setCache({ maxAge: 30 });
+
+        const config = context.environment.installation!.configuration;
+        const secret = parseJwtSecret(config.tokenSecret);
+        const token = await new jose.SignJWT({
+            sub: context.environment.installation!.id,
+        })
+            .setProtectedHeader({ alg: 'HS256' })
+            .setIssuedAt()
+            .sign(secret);
+
+        return (
+            <block>
+                <card
+                    title="Authentication token"
+                    hint="Use this token in the GitBook Terraform provider configuration."
+                >
+                    <codeblock content={token} />
+                </card>
+            </block>
+        );
+    },
+});
+
+const handleFetchEvent: FetchEventCallback<TerraformRuntimeContext> = async (request, context) => {
+    const { api } = context;
+    const authToken = (request.headers.get('Authorization') || '').match(tokenRegExp)?.[1];
+    if (!authToken) {
+        return new Response('Unauthorized', { status: 401 });
+    }
+
+    // First, parse the installation ID from the JWT *without* verifiying the
+    // signature, which we need to do so we can fetch the sign secret from the
+    // installation configuration.
```

Comment:
The benefit of this approach is that the Terraform provider can just use a fixed URL to obtain a token; e.g. no need to separately configure an installation ID.